frontend-components/plotly/src/components/Chart.tsx

```diff
@@ -0,0 +1,474 @@
+import { ICONS, DARK_CHARTS_TEMPLATE, LIGHT_CHARTS_TEMPLATE } from "./Config";
+import OverlayChartDialog from "./Dialogs/OverlayChartDialog";
+import TitleChartDialog from "./Dialogs/TitleChartDialog";
+import TextChartDialog from "./Dialogs/TextChartDialog";
+import { Icons as PlotlyIcons } from "plotly.js-dist-min";
+import { init_annotation } from "../utils/addAnnotation";
+import { PlotConfig, hideModebar } from "./PlotlyConfig";
+import autoScaling from "./AutoScaling";
+import { non_blocking, saveImage } from "../utils/utils";
+import ResizeHandler from "./ResizeHandler";
+import ChangeColor from "./ChangeColor";
+
+const Plot = createPlotlyComponent(Plotly);
+
+export default function Chart({
+  json,
+  date,
+  cmd,
+  title,
+  global,
+  info,
+}: {
+  // @ts-ignore
+  json: Figure;
+  date: Date;
+  cmd: string;
+  title: string;
+  global: any;
+  info?: any;
+}) {
+  const posthog = usePostHog();
+
+  if (info)
+    useEffect(() => {
+      if (posthog) posthog.capture("chart", info);
+    }, [posthog]);
+
+  delete json.layout.width;
+  delete json.layout.height;
+  json.layout.title.text = "";
+
+  const [globals, setGlobals] = useState(global);
+
+  const [barButtons, setModeBarButtons] = useState({});
+  const [LogYaxis, setLogYaxis] = useState(false);
+  const [chartTitle, setChartTitle] = useState(title);
+  const [axesTitles, setAxesTitles] = useState({});
+  const [plotLoaded, setPlotLoaded] = useState(false);
+  const [modal, setModal] = useState({ modal: "" });
+  const [loading, setLoading] = useState(false);
+  const [plotDiv, setPlotDiv] = useState(null);
+  const [Volume, setVolume] = useState({ old_nticks: {} });
+
+  const [plotData, setPlotData] = useState(json);
+  const [annotations, setAnnotations] = useState([]);
+  const [changeTheme, setChangeTheme] = useState(false);
+  const [darkMode, setDarkMode] = useState(true);
+  const [autoScale, setAutoScaling] = useState(false);
+  const [changeColor, setChangeColor] = useState(false);
+  const [colorActive, setColorActive] = useState(false);
+  const [onAnnotationClick, setOnAnnotationClick] = useState({});
+
+  const onClose = () => setModal({ modal: "" });
+
+  // @ts-ignore
+  function onDeleteAnnotation(annotation) {
+    console.log("onDeleteAnnotation", annotation);
+    const index = plotData?.layout?.annotations?.findIndex(
+      (a: any) => a.text === annotation.text
+    );
+    console.log("index", index);
+    if (index > -1) {
+      plotData?.layout?.annotations?.splice(index, 1);
+      setPlotData({ ...plotData });
+      setAnnotations(plotData?.layout?.annotations);
+    }
+  }
+
+  // @ts-ignore
+  function onAddAnnotation(data) {
+    init_annotation({
+      plotData,
+      popupData: data,
+      setPlotData,
+      setModal,
+      setOnAnnotationClick,
+      setAnnotations,
+      onAnnotationClick,
+      annotations,
+      plotDiv,
+    });
+  }
+
+  if (axesTitles && Object.keys(axesTitles).length > 0) {
```

Comment:
when should this run?

if it's when the component mounts it should run inside a useEffect(() => {
//code....    
},[])

Reply:
EZPZ